src/arch/armv8/aarch32/boot.S

```diff
@@ -52,11 +52,31 @@ _reset_handler:
     and r0, r0, r1
 
     /**
-     * TODO: Linearize cpu id according to the number of clusters and
-     * processors per cluster.
+     * Linearize cpu id according to the number of clusters and processors
+     * per cluster.
      */
-
     adr r1, _el2_entry
+    mov r3, r0, lsr #8
+    and r3, r3, #0xff
+    ldr r4, =platform
+    mov r5, #(PLAT_ARCH_OFF+PLAT_ARCH_CLUSTERS_OFF+PLAT_CLUSTERS_CORES_NUM_OFF)
+    ldr r4, [r4, r5]
+    ldr r5, =BAO_VAS_BASE
+    sub r4, r4, r5
+    add r4, r4, r1
+    mov r5, #0
+    mov r7, #0
+1:
+    cmp r5, r3
+    bge	2f
+    ldrb r6, [r4]
```

Comment:
Reading a byte but the array element is size_t not a uint8_t

Reply:
I've added this correct in #73. So I'd suggest only merging this after that PR is merged and @AfonsoSantos96 uses the macro in the aarch32 code also.

Reply:
#73 was merged. @AfonsoSantos96 could you re-write this PR following that fix?

Reply:
Just fixed with the changes from #73